test/plugins/adonis.spec.js

```diff
@@ -0,0 +1,342 @@
+'use strict'
+
+const axios = require('axios')
+const getPort = require('get-port')
+const agent = require('./agent')
+const plugin = require('../../src/plugins/adonis')
+const { setupResolver, Config, Logger } = require('@adonisjs/sink')
+const { ioc } = require('@adonisjs/fold')
+const path = require('path')
+
+const Server = require('@adonisjs/framework/src/Server')
+const Route = require('@adonisjs/framework/src/Route/Manager')
+const RouteStore = require('@adonisjs/framework/src/Route/Store')
+const Request = require('@adonisjs/framework/src/Request')
+const Response = require('@adonisjs/framework/src/Response')
+const Context = require('@adonisjs/framework/src/Context')
+const Exception = require('@adonisjs/framework/src/Exception')
+const BaseExceptionHandler = require('@adonisjs/framework/src/Exception/BaseHandler')
+
+wrapIt()
+
+describe('Plugin', () => {
+  let port
+
+  describe('adonis', () => {
+    withVersions(plugin, '@adonisjs/framework', version => {
+      before(() => {
+        Context.getter('request', function () {
+          return new Request(this.req, this.res, new Config())
+        }, true)
+
+        Context.getter('response', function () {
+          const config = new Config()
+          config.set('app.http.jsonpCallback', 'callback')
+          return new Response(this.request, config)
+        }, true)
+
+        setupResolver()
+        ioc.autoload(path.join(__dirname, 'app'), 'App')
+
+        return agent.load(plugin, 'adonis').then(() => {
+        })
+      })
+
+      after(() => {
+        return agent.close()
+      })
+
+      beforeEach(() => {
+        this.exception = Exception
+        this.server = new Server(Context, Route, new Logger(), this.exception)
+        this.server.bindExceptionHandler()
+
+        ioc.fake('Adonis/Exceptions/BaseExceptionHandler', () => {
+          return new BaseExceptionHandler()
+        })
+
+        return getPort().then(newPort => {
+          port = newPort
+        })
+      })
+
+      afterEach(() => {
+        RouteStore.clear()
+        ioc.restore()
+        this.exception.clear()
+        return this.server.close()
+      })
+
+      it('should do automatic instrumentation on routers', done => {
+        Route.get('/api/user', function ({ response }) {
+          return response.send('GET')
+        })
+
+        agent.use(traces => {
+          const spans = traces[0]
+
+          expect(spans[0]).to.have.property('service', 'test')
+          expect(spans[0]).to.have.property('name', 'adonis.middleware')
+          expect(spans[0].meta).to.have.property('component', 'adonis')
+
+          expect(spans[1]).to.have.property('name', 'GET')
```

Comment:
Ideally the operation name is the url w/ parameter names `/api/user`, `/api/:param1/:param2`, etc.  The method is not necessary as long as the `http.method` tag is set.